graphqlapi/src/main/resources/schemafirst/cql_directives.graphql

```diff
@@ -218,6 +218,18 @@ enum Predicate {
     CONTAINS
 }
 
+"""
+A predicate used in `@cql_if` to define a condition.
+"""
+enum IfPredicate {
```

Comment:
We don't support:
- IN, because:
```
InvalidRequest: Error from server: code=2200 [Invalid query] message="IN predicates on non-primary-key columns (firstname) is not yet supported"
```
but DELETE can target only non-primary-key

- CONTAINS: it doesn't see to work for DELETE `IF` (I was not able to do it from cqlsh)

Reply:
IN seems to work:
```
cqlsh:test> CREATE TABLE foo(k int PRIMARY KEY, v int);
cqlsh:test> INSERT INTO foo (k, v) VALUES (1, 1);
cqlsh:test> DELETE FROM foo WHERE k = 1 IF v in (1, 3);

 [applied]
-----------
      True
```

CONTAINS is indeed invalid. The CQL grammar only allows IN and the arithmetic operators ([source](https://github.com/apache/cassandra/blob/trunk/src/antlr/Parser.g#L1624-L1647)).

I wonder if it's worth creating an entirely separate predicate enum though. We could reuse `Predicate`, even if some values are not allowed in this particular context. This already happens with WHERE conditions (e.g. some predicates are not allowed for primary key fields, require a particular type of index, etc). Checking this during schema validation seems acceptable.

Reply:
I think that keeping a separate `enum` for both has a couple of important benefits; mainly, it allows GraphQL to catch the wrong condition at the parsing time. It is faster feedback, and we don't need to build the query and process it.
Also, the existing `Predicate` does not support `NEQ`, whereas the new `IfPredicate` is. Are we ok with adding the `NEQ` to the existing `Predicate`

Reply:
Regarding `IN` support. You are right, I was trying to implement it but there was an error from `QueryBuilder`. It turns out that `QueryBuilder` had a bug for the `IN` predicate. I've fixed it and implemented IN for a custom IF DELETE in:
https://github.com/stargate/stargate/pull/958/commits/eee0a1a56a0cc5b72b3e683f1075b7523afdb77c